spec/unit/shoulda/matchers/active_model/validate_length_of_matcher_spec.rb

```diff
@@ -28,6 +63,43 @@
       expect(validating_length(minimum: 0)).
         to validate_length_of(:attr).is_at_least(0)
     end
+
+    context 'when the writer method for the attribute changes incoming values' do
+      context 'and the matcher knows nothing of this' do
+        it 'raises a CouldNotSetAttributeError' do
+          pending 'this should test that name accepts values > 0 length but does not'
+
+          record = build_record_validating_length(
+            attribute_name: :name,
+            convert_attribute_with: :upcase,
+            minimum: 0
+          )
+
+          assertion = lambda do
+            expect(record).to validate_length_of(:name).is_at_least(0)
+          end
+
+          expect(&assertion).to raise_error(
+            Shoulda::Matchers::ActiveModel::AllowValueMatcher::CouldNotSetAttributeError
+          )
+        end
+      end
+
+      context 'and the matcher knows how given values get changed' do
+        it 'accepts (and not raise an error)' do
+          record = build_record_validating_length(
+            attribute_name: :name,
+            convert_attribute_with: :upcase,
+            minimum: 0
+          )
+
+          expect(record).
+            to validate_length_of(:name).
+            is_at_least(0).
+            converting_values("x" => "X")
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.